docs/feature-requests/005-editable-diff.md

```diff
@@ -0,0 +1,173 @@
+  - clicking on a new "edit hunk" button on hunk header
+  - if the hunk is selected, pressing specific keys (exact key binding TBD)
+  - "edit" from the context menu
+  - if already in the editable state in a previous hunk, navigating with the keyboard can exit previous hunk's editable state and enter the editable state of the next hunk.
+
+
+#### 2. Within an editable state
+  - Only one hunk is editable at a time
+  - Visually, it will be very clear when a hunk is in the editable state. In addition to a blinking caret, maybe we can put the whole block in a box or fade out the rest of the diff? :thinking:
+  - The editable state behaves almost like it's a normal editor, except that the green background of added lines and red background of deleted lines will be kept.
+  - A deleted line will still be visible within the editable state, but it _is not editable_. That will be communicated to users in a very clear visual manner:
+    - The cursor for hovering over deleted lines will show up as an arrow as opposed to text cursor.
+    - When navigating with the keyboard, deleted lines will be skipped as if they don't exist.
+    - When dragging the cursor to highlight a block of text, deleted lines will not get highlighted.
+
+#### 3. Exiting editable state:
+Upon exiting editable state, the changes will be immediately saved to the corresponding file on disk. And the diff view would re-render with the new file patch, and scroll position will remain unchanged.
+
+A user can exit the editable state by:
+  - clicking outside of the editable area
+  - pressing `ESC` or `cmd-s`
+  - performing actions such as stage/unstage, jump to file, etc.
+
+</details>
+
+##### Rationale for not going this route:
+- Having to explicitly toggle into an editable state is similar to behaviour seen in dotcom, but this goes against our editor-first approach. It's called an _editor_ for a reason after all -- things are expected to be editable.
+- Since [the majority of the existing diff view usage will be editable](#what-is-editable), it makes sense to default to the editable state, and only communicate to users when it's an exception (i.e. read-only).
+
+
+### Editable Split Diff
+
+All of the prior arts I could find on editable diffs implement this feature with the use of "split screen diff".
+
+![kapture 2019-01-03 at 20 35 29](https://user-images.githubusercontent.com/6842965/50657589-37134980-0f97-11e9-96cc-41cb1eda6546.gif)
+
+This is a gif of how it works in VS code, but other diff and/or merge tools have similar implementations:
+ - split screen with one side editable (the file on disk) and the other side read-only
+ - both sides show unmodified lines
+ - readonly side shows deleted lines
+ - editable side shows added/modified lines
+ - use grey/void blocks to reconcile the line differences between the two sides so they line up properly
+
+##### Pros:
+ - it's easy to understand that one side is editable, and the other is not.
```

Comment:
another pro for this approach: some users have asked for split diffs.  see https://github.com/atom/github/issues/816 and https://github.com/atom/github/issues/1423.